cloud/amazon/ec2_eni.py

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/python
+#
+# This is a free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This Ansible library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_eni
+short_description: Create and optionally attach an Elastic Network Interface (ENI) to an instance
+description:
+    - Create and optionally attach an Elastic Network Interface (ENI) to an instance. If an ENI ID is provided, an attempt is made to update the existing ENI. By passing 'None' as the instance_id, an ENI can be detached from an instance.
+version_added: "2.0"
+author: Rob White, wimnat [at] gmail.com, @wimnat
+options:
+  instance_id:
+    description:
+      - Instance ID that you wish to attach ENI to. To detach an ENI from an instance, use 'None'.
+    required: false
+    default: null 
+  private_ip_address:
+    description:
+      - Private IP address.
+    required: false
+    default: null
+  subnet_id:
+    description:
+      - Subnet in which to create the ENI. Only required when state=present.
+    required: true
+  description:
+    description:
+      - Optional description of the ENI.
+    required: false
+    default: null
+  security_groups:
+    description:
+      - List of security groups associated with the interface. Only used when state=present.
+    required: false
+    default: null
+  state:
```

Comment:
you should document possible states here with a choices: [ ...] entry, also the 'list' state is indicative of code belonging in a facts module.